plugin/src/main/java/at/helpch/chatchat/listener/ChatListener.java

```diff
@@ -54,12 +54,6 @@ public void onChat(final AsyncPlayerChatEvent event) {
             ? user.channel()
             : channelByPrefix.get();
 
-        // Ensure the user still has the channel permission, if not reset them back to the default channel
-        if (!channel.isUsableBy(user)) {
-            channel = ChatChannel.defaultChannel();
-            user.channel(channel);
-        }
```

Comment:
Instead of removing this, maybe change it a bit:

- Move the user to default channel
- Warn the user that they do not have access to the channel and have been moved to the default one.
- Cancel the chat event.